plugin/vcs/gitlab/gitlab.go

```diff
@@ -479,84 +481,111 @@ func getRoleAndMappedRole(accessLevel int32) (gitLabRole ProjectRole, bytebaseRo
 	return "", ""
 }
 
-// FetchRepositoryActiveMemberList fetch all active members of a repository
+// FetchRepositoryActiveMemberList fetches all active members of a repository.
+//
+// Docs: https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project
 func (p *Provider) FetchRepositoryActiveMemberList(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID string) ([]*vcs.RepositoryMember, error) {
-	// Official API doc: https://docs.gitlab.com/14.6/ee/api/members.html
-	url := fmt.Sprintf("%s/projects/%s/members/all", p.APIURL(instanceURL), repositoryID)
-	code, body, err := oauth.Get(
-		ctx,
-		p.client,
-		url,
-		&oauthCtx.AccessToken,
-		tokenRefresher(
-			instanceURL,
-			oauthContext{
-				ClientID:     oauthCtx.ClientID,
-				ClientSecret: oauthCtx.ClientSecret,
-				RefreshToken: oauthCtx.RefreshToken,
-			},
-			oauthCtx.Refresher,
-		),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "GET")
-	}
-
-	if code == http.StatusNotFound {
-		return nil, common.Errorf(common.NotFound, fmt.Errorf("failed to fetch repository members from GitLab instance %s", instanceURL))
-	} else if code >= 300 {
-		return nil, fmt.Errorf("failed to read repository members from GitLab instance %s, status code: %d",
-			instanceURL,
-			code,
+	// Preallocate the slice to avoid multiple allocations when appending the first
+	// 100 results to the slice.
+	allMembers := make([]RepositoryMember, 0, 100)
+
+	// The "state" filter only available in GitLab Premium self-managed, GitLab
+	// Premium SaaS, and higher tiers, but worth a try for less abandoned results.
+	nextURL := fmt.Sprintf("%s/projects/%s/members/all?state=active&per_page=100", p.APIURL(instanceURL), repositoryID)
```

Comment:
Can we follow the same pattern that we use for https://github.com/bytebase/bytebase/blob/main/plugin/vcs/gitlab/gitlab.go#L293 ?

Reply:
TIL, did not realize we already have pagination support in other methods!

Few questions though:

1. Why is the `perPage` set to 40 not 100? I do not see a comment explaining the rationale.
2. Use `if len(temp) < perPage {` would waste an API call if the last page happen to have the exact same number of `perPage`, and no next page.
3. By "same pattern", do you mean having a helper like `p.fetchOffsetRepositoryList` or something else?

Reply:
1. don't matter :-) can be 40 or 100 as long as things are consistent.
2. That's okay to waste a call if that's simpler and more readable. We may want to avoid using link.Parse(header.Get("Link"))["next"] which introduced an additional library/dependency.
3. I mean the coding pattern should be consistent.

Reply:
> 3\. I mean the coding pattern should be consistent.

Could you be more specific and/or give some examples?

Reply:
I mean "copy&paste" the following regardless. Or change both if there's better pattern.
https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/plugin/vcs/gitlab/gitlab.go?L289&subtree=true

Reply:
Addressed in https://github.com/bytebase/bytebase/pull/1423/commits/cc6f42868e8f2b6e7757c07b13fb68e6107ac470